src/BuildScriptGenerator/Node/NodeLanguageDetector.cs

```diff
@@ -149,5 +154,51 @@ private string DetectNodeVersion(dynamic packageJson)
 
             return nodeVersion;
         }
+
+        private bool IsStaticSite(ISourceRepo sourceRepo)
+        {
+            return IsHugoSite(sourceRepo);
+        }
+
+        private bool IsHugoSite(ISourceRepo sourceRepo)
+        {
+            // Hugo configuration variables: https://gohugo.io/getting-started/configuration/#all-configuration-settings
+            var hugoVariables = new[] { "baseURL", "title", "languageCode", "theme" };
+            var lines = new string[] { };
+
+            // Search for config.toml or config/config.toml
+            if (sourceRepo.FileExists(NodeConstants.HugoTomlFileName))
+            {
+                lines = sourceRepo.ReadAllLines(NodeConstants.HugoTomlFileName);
+            }
+            else if (sourceRepo.FileExists(NodeConstants.HugoConfigFolderName, NodeConstants.HugoTomlFileName))
+            {
+                lines = sourceRepo.ReadAllLines(NodeConstants.HugoConfigFolderName, NodeConstants.HugoTomlFileName);
+            }
+
+            if (lines.Any(l => hugoVariables.Any(v => l.TrimStart(' ').StartsWith(v))))
+            {
+                return true;
+            }
+
+            // Search for config/*.toml.
+            if (sourceRepo.DirExists(NodeConstants.HugoConfigFolderName))
```

Comment:
nit: just a general guideline for readability...to avoid deep nested code, you can short-circuit the code early (ex: return false if directory does not exist)